Table data full viewer fix

will-moore opened the pull request:
# What this PR does

Fixes a bug reported in
https://www.openmicroscopy.org/community/viewtopic.php?f=4&t=8345
where the Tables data is not displayed in the full image viewer.

# Testing this PR

1. Create tables data on Sreen AND on a Plate, with rows corresponding to Wells, as described in http://help.openmicroscopy.org/scripts.html#metadata
1. Confirm that this data is shown in webclient right panel under Tables tab when Well is selected
1. Open images from that Well in full image viewer and check that Tables data is shown under "Image Information" popup

<img width="414" alt="screen shot 2017-08-16 at 23 08 38" src="https://user-images.githubusercontent.com/900055/29387460-021edbb0-82d8-11e7-96fd-918223062625.png">

TODO:
 - [ ] Add robot test

will-moore committed:
imageMarshal imgData json allows wellId AND datasetId

will-moore committed:
Fix query for tables lookup from image viewer

eleanorwilliams commented:
I tried to test this on http://web-dev-merge.openmicroscopy.org/webclient/ logged in at first as user-4 and then as user-1, but although I could add a csv file to a plate or screen, the populate metadata script failed to create a bulk annotations file and gives an error   `Unable to create table: %s" % name)
omero.util.populate_metadata.MetadataError: Unable to create table: bulk_annotations`. 

I couldn't find any plates with tables already created.

will-moore commented:
Even Plates with ```bulk_annotations``` attached don't show any Tables data for their Wells. E.g. http://web-dev-merge.openmicroscopy.org/webclient/?show=plate-2072 (user-3)
Not sure what's going on here: Locally I also can't create tables but display of existing tables data is OK.

will-moore commented:
Table data is now being displayed ok at http://web-dev-merge.openmicroscopy.org/webclient/?show=well-336745 (user-3) and in Full Viewer: http://web-dev-merge.openmicroscopy.org/webclient/img_detail/51637/

jburel commented:
Table data displayed
The display could be improved in the viewer but not worth spending time on it.
To come robot tests

will-moore committed:
Remove unused omeTiffDisabled check in toolbar.html

Noticed during code review for https://www.openmicroscopy.org/community/viewtopic.php?f=6&t=8338
that this variable is never set in context, but it's ignored because 'not omeTiffDisabled'
always evaluates as True

will-moore committed:
Robot test for bulk annotations in viewer

jburel commented on components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/annotations/includes/toolbar.html:
```diff
@@ -205,7 +205,7 @@
 
             {% if not share %}
-            {% if image and not omeTiffDisabled %}
+            {% if image %}
```
This commit is not related to the description of the PR
Why did you include it?

will-moore replied:
It seemed so minor as to not justify a separate PR, but I can put it in a separate PR?

jburel replied:
in the "export" PR discussed this am

jburel replied:
Could you put the commit in another PR since the change is actually not tested the same way

will-moore commented:
Robot test is failing because previous test is leaving the Image Viewer window open, not the webclient.
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/691/robot/Web%20&%20Web/Web_1/View%20Image/Test%20Bulk%20Annotations/
Will fix...

will-moore committed:
Fix robot test

will-moore commented:
Passing now: https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/692/robot/Web%20&%20Web/Web_1/View%20Image/Test%20Bulk%20Annotations/

mtbc commented:
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/693/robot/Web%20&%20Web/Web/View%20Image/Test%20Bulk%20Annotations/ is still good.

will-moore committed:
Revert "Remove unused omeTiffDisabled check in toolbar.html"

Removing commit from PR #5441.
This reverts commit 776f5dc873fc2b41e3d5851d1383a989cd41cabf.

jburel commented:
Thanks for rolling back the commit
